math_notes/mathnotes.py

```diff
@@ -0,0 +1,208 @@
+
+        Event here is where the mouse was and is located. Also draws the same line
+        on a PIL image which represents the image drawn on the tkinter canvas by the
+        user. This is the image that will actually be saved and used by the OCR.
+
+        :param event: Recorded mouse-click events for drawing.
+        :type event: class tkinter.Event
+
+        :return: none
+        :rtype: none
+        """
+
+        # This canvas call is what the user sees on the screen.
+        canvas.create_line(
+            (lastx, lasty, event.x, event.y),
+            smooth=True,
+            width=linewidth,
+            fill=linecolor,
+        )
+
+        # The draw call is in the background (invisible)
+        # capturing what will actually get converted to an image.
+        draw.line(
+            [lastx, lasty, event.x, event.y],
+            fill=linecolor,
+            width=linewidth,
+            joint="curve",
+        )
+        _save_posn(event)
+
+    global lastx, lasty
+
+    root = Tk()
+    root.title("Math Canvas")
+
+    text = tk.StringVar()
+    text.set("Prediction (ready): ")
+
+    # Define the supporting paths.
+    filename["path"] = filename.get("path", Path("./math_notes/temp_files/"))
+
+    # Create the canvas.
+    canvas_dimensions = str(width) + "x" + str(int(height * 1.5))
+    root.geometry(canvas_dimensions)
+
+    # Instantiate the tkinter canvas for users to draw on.
+    canvas = Canvas(root, bg="white", width=width, height=height)
+    canvas.grid(row=0, column=0)
+
+    # PIL create an empty image and draw object to memory only.
+    # It is not visible.
+    canvas_image = Image.new("RGB", (width, int(height * 1.5)), (255, 255, 255))
+    draw = ImageDraw.Draw(canvas_image)
+
+    # Capturing mouse motion.
+    canvas.bind("<Button-1>", _save_posn)
+    canvas.bind("<B1-Motion>", _add_line)
+
+    # Labels for readouts.
+    row = 10
+    label = tk.Label(root, textvariable=text, font=("helvetica", 18))
+    label.grid(row=row + 3, column=0)
+
+    # Buttons to save canvas, quit canvas, browse images.
+    myFont = font.Font(family="SFMono-Regular", size=11)
+
+    button_explore = Button(
+        root,
+        text="Select a file",
+        command=lambda: be._browse_files(filename),
+        height="2",
+        width="12",
+        font=myFont,
+    )
+
+    button_predict = Button(
+        text="Predict LaTeX!",
+        command=lambda: be._ocr_request_button(
+            filename, canvas_image=canvas_image, text=text
+        ),
+        height="2",
+        width="12",
+        font=myFont,
+    )
+
+    button_quit = Button(
+        text="Quit", command=lambda: be._quit(root), height="2", width="12", font=myFont
+    )
+
+    button_predict.grid(row=row, column=0)
+    button_explore.grid(row=row + 1, column=0)
+    button_quit.grid(row=row + 2, column=0)
+
+    # Execute tkinter commands.
+    root.mainloop()
+
+    return _latex_read()
+
+
+@register_line_magic
+def mn(line):
```

Comment:
This is not a great function name. What the hell does `mn` even mean? If it's main, call it `main`. An extra 2 letters won't kill you, and adds a lot to readability, which is among the most important traits of good code.

Reply:
I wanted a good name for using mathnotes quickly so I chose two letters for the magic command. Before I used 'mathnotes' or similar, but I want to keep it to something short.

Reply:
I decided to name this mathnote, since there is one note on this line to be called in the cell.